api/migrations/20200603153526-bcmiDataLoad.js

```diff
@@ -0,0 +1,135 @@
+    mClient = await db.connection.connect(db.connectionString, { native_parser: true });
+    const nrpti = mClient.collection('nrpti');
+
+    // API call to pull data from BCMI
+    // could also fetch from NRPTI first: require('../src/models/bcmi/mine').find().then(...);
+    // then match to BCMI, just doing it this way so we can match on name, rather then code
+    console.log('Fetching all major mines in BCMI...');
+    // 30 results for major, 74 for published. Note, published does not return links so if we're supposed to use
+    // published, we'll need to do a follow up call to /api/project/bycode/<mineData.code> to get the links
+    const publishedMines = await getRequest(bcmiUrl + '/api/projects/published');
+    console.log('Located ' + publishedMines.length + ' mines. Batching updates...');
+
+    const promises = [];
+    // build up a collection of all requests
+    for (let i = 0; i < publishedMines.length; i++) {
+      try {
+        // The published endpoint doesn't have links, refresh the mineData object
+        let mineData = await getRequest(bcmiUrl + '/api/project/bycode/' + publishedMines[i].code);
+        promises.push(updateMine(mineData, nrpti));
+      } catch(err) {
+        console.error('Could not find ' + publishedMines[i]._id + ' : ' + publishedMines[i].name);
+        // dont rethrow, we'll just ignore this one as a failure and check the rest
+      }
+    }
+
+    // fire off the requests and wait
+    let results = await Promise.all(promises);
+
+    let updatedCount = 0;
+    let notFoundCount = 0;
+    results.forEach(result => {
+      if (Object.prototype.hasOwnProperty.call(result, 'notfound')) {
+        notFoundCount++;
+        console.log('Could not find ' + result.data._id + ' : ' + result.data.name);
+      } else {
+        updatedCount++;
+      }
+    });
+
+    // we're done
+    console.log('BCMI migration complete.');
+    console.log('Of ' + publishedMines.length + ' mines in BCMI, ' + notFoundCount + ' could not be found in NRPTI, and ' + updatedCount + ' were updated.');
+  } catch(err) {
+    console.error('Error on BCMI dataload: ' + err);
+  }
+
+  mClient.close();
+};
+
+exports.down = function(db) {
+  return null;
+};
+
+exports._meta = {
+  "version": 1
+};
+
+async function updateMine(mineData, nrpti) {
+  let nrptiMines = await nrpti.find({ _schemaName: 'Mine', name: mineData.name}).toArray();
+  // should have 1 result returned. Any more or less, just ignore this update
+  if (nrptiMines.length === 1) {
+    let externalLinks = [];
+    // format the links from the data
+    for(const idx in mineData.externalLinks) {
+      const link = mineData.externalLinks[idx];
+      externalLinks.push(link.link);
+    }
+
+    return nrpti.update({ _id: new ObjectID(nrptiMines[0]._id) }, {
+      $set: {
+        type:            mineData.type,
+        summary:         mineData.description, // BCMI doesn't have a "summary" attribute
+        description:     mineData.description,
```

Comment:
Not a PR issue, just some context:

The reason we have summary and description in some places, is because LNG uses description, but NRCED uses summary.  The reason being, is that NRCED internally uses both description and summary, but not in the same way LNG does.  So we ended up supporting both, depending on which flavour you're looking at. 

So if Mines also use both, then this is totally fine.  If not, then we'll probably end up removing one of them later on as we refine things.

Reply:
Alright, good to know. I'm expecting this might be extended for document support in the short term as well, so we can readdress the model as needed.